### DISCUSSION TITLE
Should custom panel styles be overridden by defaults?

### DISCUSSION THREAD

**Discussion by nicholaskillin** [TanStack/query#5940] [General]
We recently made a change to our app that caused the right side of the devtools panel to be covered by another element. My plan was to just pass in a custom width via a style object supplied to `panelProps`. I was a little surprised to discover that my width was not being applied.

After some digging it looks like when `panelStyle` is passed into the `getSidePanelStyle` function those styles get immediately overwritten by any of the defaults defined after those styles are spread.

https://github.com/TanStack/query/blob/1c2171755ba835b3fac603228ff64436fff9c28d/packages/react-query-devtools/src/utils.ts#L226-L242

I guess my question is, is that intentional? It seem to me that the common practice when providing flexibility like this is to assume that custom supplied styles will override the defaults.

Happy to open a PR if the general consensus is that styles provided via `panelProps` should override all defaults.

**Comment by TkDodo** [comment]
sounds like a bug. do you want to fix it?

**Reply by TkDodo** [reply]
oh, I forgot: In v5 (beta), we have new devtools. Can you take a look if it works there, and if not, maybe only fix it there, or we'd need to port it from the react specific to the agnostic implementation ...

**Reply by nicholaskillin** [reply]
Yeah, I can look to see if it works there, although I'm not familiar with the new V5 beta stuff at all. Should I be looking in the `beta` branch for the latest v5 implementation?

**Reply by TkDodo** [reply]
yes please

**Reply by nicholaskillin** [reply]
So I did some digging just now in the beta code and it is a little less clear to me if this is still an issue. It seems that the `panelProps` prop for `ReactQueryDevtools` was removed. Is there no way to customize the style in V5, or was the method for tweaking it changed in a way that I haven't discovered yet?

**Reply by TkDodo** [reply]
@ardeora maybe ?

**Reply by ardeora** [reply]
Yes, I want to provide a way to allow devtools to allow custom styles. Will adding a custom class to each devtools element help?

For example the devtools panel can have a className like `TQDevtools-Panel`. The query rows can have a custom class like `TQDevtools-QueryRow` and so on...

We can have all the elements to have a prefixed class name starting with `TQDevtools-` so that users can target any element they would like to customize.

**Reply by nicholaskillin** [reply]
Yeah, that would work just fine for our situation.

@TkDodo Do you still want me to open a PR to fix this in V4 or are we no longer concerned with updating that version since 5 is in beta?

**Reply by TkDodo** [reply]
I'd prefer to only merge things for v5 now please.
